Rewrote a lot of the usage documentation (#259)
nf-core/nf-co.re: markdown/usage/introduction.md

Whether your pipeline is a simple BLAST execution or a complex genome annotation pipeline, you can build it with Nextflow.

## How to run a pipeline

> Nextflow works best when you have an active internet connection, as it is able to fetch all pipeline reqiurements. If you need to run offline, please see [_Running offline_](offline.md).

Reviewer: ```suggestion
Nextflow works best when you have an active internet connection, as it is able to fetch all pipeline requirements. If you need to run offline, please see [_Running offline_](offline.md).
```

1. First, make sure that you have all required software installed (Nextflow + Docker / Singularity / Conda). See the [installation docs](installation.md) for more information.

2. Choose a pipeline to run. See the available pipelines at [https://nf-co.re/pipelines](https://nf-co.re/pipelines). If you have [nf-core/tools](tools.md) installed, run `nf-core list`

3. Configure Nextflow to run on your system.

>     * The simplest way to run is with `-profile docker` (or `singularity` /  `conda`) which will tell nextflow will execute jobs locally using Docker for software requirements.

Reviewer: ```suggestion
    * The simplest way to run is with `-profile docker` (or `singularity` /  `conda`) which will tell Nextflow to execute jobs locally using Docker to fulfil the software requirements.
```

>     * For more complex configuration of nextflow for your system, please see the [_Nextflow configuration_](https://nf-co.re/usage/configuration) documentation.

Reviewer: ```suggestion
    * For more complex configuration of Nextflow for your system, please see the [_Nextflow configuration_](https://nf-co.re/usage/configuration) documentation.
```

> 4. To test that everything is working properly, try running the tests for you your pipeline of interest in your terminal:

Reviewer: ```suggestion
4. To test that everything is working properly, try running the tests for your pipeline of interest in the terminal:
```

    ```bash
    nextflow run nf-core/<pipeline_name> -profile test,docker
    ```

>     * Replace `<pipeline_name>` with the name of a nf-core pipeline.

Reviewer: ```suggestion
    * Replace `<pipeline_name>` with the name of an nf-core pipeline.
```

    * There is no need to download anything first - nextflow will pull the code for you from the GitHub repository automatically and fetch the software requirements too.

>     * If you don't have Docker installed, replace `docker` in the command with either `singularity` or `conda`.

Reviewer: Move this point up to 2nd in the listing.

>     * If the pipeline fails, check the [troubleshooting docs](/usage/troubleshooting.md) and ask for help on the nf-core Slack (see [https://nf-co.re/join](https://nf-co.re/join)).

Reviewer: ```suggestion
    * If the pipeline fails, check the [troubleshooting docs](/usage/troubleshooting.md) and ask for help on the nf-core Slack channel for that particular pipeline (see [https://nf-co.re/join](https://nf-co.re/join)).
```

5. Read the pipeline documentation to see which command-line parameters are required. These will be specific to your data type and usage.

6. Launch the pipeline with some real data by omitting the `test` config profile and providing the required pipeline-specific parameters. For example, if you want to run the `methylseq` pipeline, you might use the following command:

    ```bash
    nextflow run nf-core/methylseq -profile docker -reads 'input_data/*.fastq.gz' --outdir myproj/results --genome GRCh38
    ```

7. Once complete, check the pipeline execution and quality control reports. Each pipeline comes with documentation describing the different outputs.

## Tips and tricks

> * Hyphens matter! Core nextflow options use one (`-`), pipeline specific paramters use two (`--`)

Reviewer: ```suggestion
* Hyphens matter! Core Nextflow command-line options use one (`-`) whereas pipeline specific parameters use two (`--`)
```

* Specify `--email your@email.com` to receive emails when your pipeline completes
* Always specify `-r <version-number>` when running to explicitly use a specific release. Then an identical command can be used in the future to give identical results.

> * Use `-resume` to restart pipelines that did not complete: successful tasks skipped using the outputs from the previous run.

Reviewer: ```suggestion
* Use `-resume` to restart pipelines that did not complete. This ensures that successful tasks from the previous run wont be re-executed.
```

        ```bash
        nextflow run hello
        ```

2. Choose a pipeline to run. See the available pipelines at [https://nf-co.re/pipelines](https://nf-co.re/pipelines). If you have [nf-core/tools](tools.md) installed, run `nf-core list`

3. Configure Nextflow to run on your system.

    * The simplest way to run is with `-profile docker` (or `singularity` /  `conda`) which will tell Nextflow to execute jobs locally using Docker to fulfil the software requirements.

    * For more complex configuration of Nextflow for your system, please see the [_Nextflow configuration_](https://nf-co.re/usage/configuration) documentation.

4. To test that everything is working properly, try running the tests for your pipeline of interest in the terminal:

    ```bash
    nextflow run nf-core/<pipeline_name> -profile test,docker
    ```

    * Replace `<pipeline_name>` with the name of an nf-core pipeline.

    * If you don't have Docker installed, replace `docker` in the command with either `singularity` or `conda`.

    * There is no need to download anything first - nextflow will pull the code for you from the GitHub repository automatically and fetch the software requirements too.

>     * If the pipeline fails, check the [troubleshooting docs](/usage/troubleshooting.md) and ask for help on the nf-core Slack channel for that particular pipeline (see [https://nf-co.re/join](https://nf-co.re/join)).

Reviewer 2: ```suggestion
    * If the pipeline fails, check the [troubleshooting docs](troubleshooting.md) and ask for help on the nf-core Slack channel for that particular pipeline (see [https://nf-co.re/join](https://nf-co.re/join)).
```
I think we're already in `usage/` here in `introduction.md`, so the link should be `troubleshooting.md`

Author: True. Should work on the website, as the first `/` sets the path to the web root. But it'll probably break on GitHub.

5. Read the pipeline documentation to see which command-line parameters are required. These will be specific to your data type and usage.

6. Launch the pipeline with some real data by omitting the `test` config profile and providing the required pipeline-specific parameters. For example, if you want to run the `methylseq` pipeline, you might use the following command:

    ```bash

>     nextflow run nf-core/methylseq -profile docker -reads 'input_data/*.fastq.gz' --outdir myproj/results --genome GRCh38

Reviewer 2: ```suggestion
    nextflow run nf-core/methylseq -profile docker --reads 'input_data/*.fastq.gz' --outdir myproj/results --genome GRCh38
```
